pkg/storage/elasticsearch.go

```diff
@@ -122,11 +123,30 @@ func (ed *ElasticsearchDeployment) InjectStorageConfiguration(p *corev1.PodSpec)
 
 // InjectSecretsConfiguration changes the given spec to include the options for the index cleaner
 func (ed *ElasticsearchDeployment) InjectSecretsConfiguration(p *corev1.PodSpec) {
+	secretName := curatorSecret.instanceName(ed.Jaeger)
```

Comment:
I thought that the logic should by default use `jaeger` secret and fallback to the `curator`  if the env var is enabled (no matter if rollover is used)

Reply:
Then the community driven upstream operator would not work. There the writing of dependencies cannot be decativated. Since its missing https://github.com/jaegertracing/jaeger/commit/3499c88f640e8d9c54cde37c7b0de32c98bc80e5.

Reply:
There is no community/alternative for OpenShift Elasticsearch.

Reply:
I mean upstream Jaeger in OpenShift using OpenShift Elasticsearch.

Reply:
The `es` deployment ends in a `CrashLoopBackOff` with:
```
[2022-06-07 12:00:44,199][INFO ][container.run            ] Removing unnecessary JKS files
cp: cannot create regular file '/etc/elasticsearch/elasticsearch.yml': Permission denied
cp: cannot create regular file '/etc/elasticsearch/log4j2.properties': Permission denied
```

Reply:
Still facing some issues. Any idea what could be wrong @pavolloffay @rubenvp8510 ?

```
Type     Reason       Age                   From               Message
----     ------       ----                  ----               -------
Normal   Scheduled    12m                   default-scheduler  Successfully assigned bb-testing/tenant1-query-6d6c9b6d69-jbzqh to ocp410-jk-s96nv-worker-0-tg9ld
Warning  FailedMount  12m                   kubelet            MountVolume.SetUp failed for volume "tenant1-ui-oauth-proxy-tls" : secret "tenant1-ui-oauth-proxy-tls" not found
```

```bash
$ oc get secrets -n bb-testing | grep tenant1-ui-oauth-proxy-tls
# getting
NAME                               TYPE                                  DATA   AGE
tenant1-ui-oauth-proxy-tls         kubernetes.io/tls                     2      12m
```

```yaml
apiVersion: logging.openshift.io/v1
kind: Elasticsearch
metadata:
  annotations:
    logging.openshift.io/elasticsearch-cert-management: "true"
    logging.openshift.io/elasticsearch-cert.jaeger-shared-es: user.jaeger
    logging.openshift.io/elasticsearch-cert.curator-shared-es: system.logging.curator
  name: shared-es
  namespace: bb-testing
spec:
  managementState: Managed
  nodeSpec:
    resources:
      limits:
        memory: 1Gi
      requests:
        cpu: 200m
        memory: 1Gi
  nodes:
    - nodeCount: 1
      proxyResources: {}
      resources: {}
      roles:
        - master
        - client
        - data
      storage: {}
  redundancyPolicy: ZeroRedundancy
---
apiVersion: jaegertracing.io/v1
kind: Jaeger
metadata:
  name: tenant1
  namespace: bb-testing
spec:
  strategy: production
  query:
    strategy:
      type: RollingUpdate
  collector:
    strategy:
      type: RollingUpdate
  ingress:
    openshift:
      sar: "" # allow login for all OCP users for testing
  storage:
    type: elasticsearch
    options:
      es:
        index-prefix: tenant1
    esIndexCleaner:
      enabled: true
      schedule: "*/2 * * * *"
      numberOfDays: 1
    esRollover:
      schedule: "*/1 * * * *"
      conditions: '{"max_age":"10m"}'
      readTTL: 10m
    elasticsearch:
      name: shared-es
      doNotProvision: true
      nodeCount: 1
      resources:
        requests:
          cpu: 200m
          memory: 1Gi
        limits:
          memory: 1Gi
```

Reply:
Hard to tell from the logs above.

Could you please paste here all the secrets created in the namespace where the ES instance is created?

Reply:
Sure :)

```
$ oc get secrets -n bb-testing
NAME                               TYPE                                  DATA   AGE
builder-dockercfg-cdp48            kubernetes.io/dockercfg               1      4h13m
builder-token-mj6xd                kubernetes.io/service-account-token   4      4h13m
builder-token-zgnjg                kubernetes.io/service-account-token   4      4h13m
curator-shared-es                  Opaque                                3      3h11m
default-dockercfg-2vpv4            kubernetes.io/dockercfg               1      4h13m
default-token-d6f58                kubernetes.io/service-account-token   4      4h13m
default-token-nmfsk                kubernetes.io/service-account-token   4      4h13m
deployer-dockercfg-s2dfl           kubernetes.io/dockercfg               1      4h13m
deployer-token-clhqh               kubernetes.io/service-account-token   4      4h13m
deployer-token-pzklx               kubernetes.io/service-account-token   4      4h13m
elasticsearch                      Opaque                                7      174m
jaeger-shared-es                   Opaque                                3      3h11m
shared-es                          Opaque                                7      3h11m
shared-es-dockercfg-dsf6s          kubernetes.io/dockercfg               1      3h11m
shared-es-metrics                  kubernetes.io/tls                     2      3h11m
shared-es-token-64rhd              kubernetes.io/service-account-token   4      3h11m
shared-es-token-xcshp              kubernetes.io/service-account-token   4      3h11m
signing-shared-es                  Opaque                                3      3h11m
tenant1-collector-headless-tls     kubernetes.io/tls                     2      174m
tenant1-curator                    Opaque                                3      174m
tenant1-dockercfg-rshp7            kubernetes.io/dockercfg               1      174m
tenant1-jaeger-elasticsearch       Opaque                                3      174m
tenant1-master-certs               Opaque                                2      174m
tenant1-token-qcpdd                kubernetes.io/service-account-token   4      174m
tenant1-token-wkcwc                kubernetes.io/service-account-token   4      174m
tenant1-ui-oauth-proxy-tls         kubernetes.io/tls                     2      174m
tenant1-ui-proxy-dockercfg-5dccc   kubernetes.io/dockercfg               1      174m
tenant1-ui-proxy-token-5cz99       kubernetes.io/service-account-token   4      174m
tenant1-ui-proxy-token-w7qlx       kubernetes.io/service-account-token   4      174m
```

Reply:
they all do not work.
```bash
Events:
  Type     Reason       Age                     From     Message
  ----     ------       ----                    ----     -------
  Warning  FailedMount  141m (x28 over 7h18m)   kubelet  Unable to attach or mount volumes: unmounted volumes=[certs], unattached volumes=[certs kube-api-access-wtqvx tenant1-ui-oauth-proxy-tls tenant1-service-ca tenant1-ui-configuration-volume tenant1-trusted-ca]: timed out waiting for the condition
  Warning  FailedMount  96m (x22 over 7h33m)    kubelet  Unable to attach or mount volumes: unmounted volumes=[certs], unattached volumes=[tenant1-trusted-ca certs kube-api-access-wtqvx tenant1-ui-oauth-proxy-tls tenant1-service-ca tenant1-ui-configuration-volume]: timed out waiting for the condition
  Warning  FailedMount  75m (x22 over 7h49m)    kubelet  Unable to attach or mount volumes: unmounted volumes=[certs], unattached volumes=[kube-api-access-wtqvx tenant1-ui-oauth-proxy-tls tenant1-service-ca tenant1-ui-configuration-volume tenant1-trusted-ca certs]: timed out waiting for the condition
  Warning  FailedMount  41m (x43 over 7h40m)    kubelet  Unable to attach or mount volumes: unmounted volumes=[certs], unattached volumes=[tenant1-ui-configuration-volume tenant1-trusted-ca certs kube-api-access-wtqvx tenant1-ui-oauth-proxy-tls tenant1-service-ca]: timed out waiting for the condition
  Warning  FailedMount  21m (x28 over 7h47m)    kubelet  Unable to attach or mount volumes: unmounted volumes=[certs], unattached volumes=[tenant1-service-ca tenant1-ui-configuration-volume tenant1-trusted-ca certs kube-api-access-wtqvx tenant1-ui-oauth-proxy-tls]: timed out waiting for the condition
  Warning  FailedMount  16m (x38 over 7h38m)    kubelet  Unable to attach or mount volumes: unmounted volumes=[certs], unattached volumes=[tenant1-ui-oauth-proxy-tls tenant1-service-ca tenant1-ui-configuration-volume tenant1-trusted-ca certs kube-api-access-wtqvx]: timed out waiting for the condition
  Warning  FailedMount  113s (x239 over 7h51m)  kubelet  MountVolume.SetUp failed for volume "certs" : secret "tenant1-jaeger-shared-es" not found
```

Reply:
A couple of things I noticed:

- I had to modify a little bit the CR and set `useCertManager` to `true`, because that is used by the operator to inject the right secrets, you can see the logic here: https://github.com/jaegertracing/jaeger-operator/blob/main/pkg/storage/elasticsearch.go#L258

This is the modification I did:
```
   elasticsearch:
      name: shared-es
      doNotProvision: true
      nodeCount: 1
      useCertManagement: true
      resources:
        requests:
          cpu: 200m
          memory: 1Gi
        limits:
          memory: 1Gi
```

- I had to modify a little bit how the secret name for the curator is resolved, we use a different name when the secret is created by the Jaeger operator vs ES auto-provisioning certificates. 
 
 You can see an ugly patch here: https://github.com/rubenvp8510/jaeger-operator/commit/e16c4eae5f414673d44d146e443ab63c71dfbd5c

Note how we derived the name different, in our original approach we create a secret named `tenant1-curator`, when auto-provisioning cert is used, the ES operator creates `curator-es-shared` which is the concatenation of `curator-<ES instance name>`. May be we can use the same way for naming our secrets?

Reply:
@rubenvp8510 i tried to change the naming. But for some reason everything else broke.